### PR TITLE
Use and apply cmake-format

### LINKED ISSUE #716: Format cmake
Similar to clang-format, just set a format file for cmake and be done with it.

### PR DESCRIPTION
**Pull request by berquist**
Closes #716

This uses `cmake-format` ([here](https://pypi.org/project/cmake-format/), [here](https://github.com/cheshirekow/cmakelang)) with its default settings to enforce CMake formatting.  The config was created to include help info for each setting, but it can be removed.

- There is a single commit (2ed33046eabe161bb62627b087d42a37a7113dbc) that applies these changes.  It is ignored in the relatively standardized `.git-blame-ignore-revs` so that blaming on GitHub and [the command line](https://www.stefanjudis.com/today-i-learned/how-to-exclude-commits-from-git-blame/#add-ignored-revisions-to-your-local-git-config) can skip these non-logic changes.
- It adds a `cmake-format-test.sh` that is equivalent to `clang-format-test.sh`.
- It adds a [GitHub Action](https://github.com/marketplace/actions/cmake-format-lint-action) workflow that works the same as the one for `clang-format`.

### CODE REVIEW

**Review by jpkenny — COMMENTED** [review]
I think it's better to fence the copyright statements so we don't have one format for cmake files and a different format for other files:

# ~~~
# This comment is fenced
#   and will not be formatted
# ~~~

File: .cmake-format.yaml
Comment (jpkenny) [inline .cmake-format.yaml:1]: Probably better to make this a regular file in e.g. experimental/cmake and then explicitly invoke cmake-format with that regular file as an argument. More clear how it all works to the next developer that comes along that way.

**Comment by berquist** [comment]
I saw there is also `--first-comment-is-literal` but I prefer the fences, otherwise you can keep adding comment lines to the top block and they won't be formatted.

**Comment by jpkenny** [comment]
> I saw there is also `--first-comment-is-literal` but I prefer the fences, otherwise you can keep adding comment lines to the top block and they won't be formatted.

Agreed, I think fences are better.

**Review by jpkenny — CHANGES_REQUESTED** [review]
Everything besides the dot file looks good.

File: experimental/.cmake-format.yaml
[the remark this answers is not in the document]
Comment (berquist) [reply experimental/.cmake-format.yaml:1]: Is there a reason why?  That makes it inconsistent with pretty much every other config file in SST and the world.

[the remark this answers is not in the document]
Comment (jpkenny) [reply experimental/.cmake-format.yaml:1]: Avoid needlessly obscuring how things work. But .clang-format is already a dot file, so it's fine.

**Review by jpkenny — APPROVED** [review]
Thanks for the changes, looks good.

**Pull request merged**